libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/ManeuverProcessor.kt

```diff
@@ -152,7 +158,13 @@ internal object ManeuverProcessor {
                         steps[stepIndex].bannerInstructions()?.let { bannerInstruction ->
                             val maneuverList = mutableListOf<Maneuver>()
                             bannerInstruction.forEach { banner ->
-                                maneuverList.add(transformToManeuver(banner, distanceFormatter))
+                                val maneuver = transformToManeuver(banner, distanceFormatter)
+                                if (maneuverList.find {
+                                    it.primary.text.contains(YOUR_DESTINATION_WILL, true)
+                                } == null
+                                ) {
```

Comment:
Cannot think of anything better. 

#### The above logic fixes 

list[n-1] = "Your destination will be on the left/right
list[n] = "Your destination is on the right/left"
In this case I remove the last one.

#### The above logic does not fixes 

list[n-1] = "You will arrive at your destination"
list[n] = "You have arrived at your destination"

String comparison will not work because it's difficult to predict what kind of text can be received in the directions response.

cc @LukasPaczos

Reply:
This sounds fine to me :+1:

Reply:
"You will arrive at your destination" and "You have arrived at your destination" are separate instructions, we probably shouldn't consider them duplicates.

Reply:
Wait, I'm looking at the wrong thing. I don't think we should be filtering these out at all, should we?

"Your destination will be on the left/right" and "Your destination is on the right/left" are also different instructions.

Reply:
Yes they are, but I have heard from different developers that both these instructions doesn't make sense right next to each other.

Reply:
Now that you said it out loud, I am thinking there can be various different ways directions can send these arrival instructions and it will be difficult for us to predict the string comparison. As long as the instructions are different 
- "Your destination will..." and "Your destination is..."
- "You will arrive..." and "You have arrived..."
there is no need to remove. If customers want they can filter them out or replace them with other instructions.

Wdyt? cc @LukasPaczos

Reply:
That sounds more like a customization option for the Directions API rather than the SDK. We would play 2 different voice instructions here as well.

The story is different when instruction is the same and only the distance is different, but these actually have different text.

What if we leave this one as is and recommend developers filter them out if they need to? That is possible with the current API.

Reply:
I see we commented at the same time and agreeing to the same thing :)

Reply:
😄 sounds good to me. Removing other duplicate instructions should be enough at this point and we can look into working on the arrival instructions based on customer reactions later.